sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuDeviceManager.scala

```diff
@@ -301,7 +301,9 @@ object GpuDeviceManager extends Logging {
         Rmm.initialize(init, logConf, poolAllocation)
         RapidsBufferCatalog.init(conf)
       } catch {
-        case e: Exception => logError("Could not initialize RMM", e)
+        case e: CudfException =>
+          logError("Could not initialize RMM, exiting!", e)
```

Comment:
This is typically an anti-pattern

https://stackoverflow.com/questions/6639963/why-is-log-and-throw-considered-an-anti-pattern

Is there a reason we need the log statement? Should we just lat the exception continue up?

Reply:
I would rather have the code that exits say we are exiting, otherwise there is coupling between the code that exits and here for no good reason.

Reply:
@revans2 handled here: https://github.com/NVIDIA/spark-rapids/pull/5243/commits/f6545d334ecf1dd7e496e12b5212bbc7afe1fc5f